packages/opencensus-web-instrumentation-zone/test/test-interaction-tracker.ts

```diff
@@ -0,0 +1,408 @@
+/**
+ * Copyright 2019, OpenCensus Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import 'zone.js/dist/zone.js';
+import { tracing, Span } from '@opencensus/web-core';
+import {
+  InteractionTracker,
+  RESET_TRACING_ZONE_DELAY,
+} from '../src/interaction-tracker';
+
+describe('InteractionTracker', () => {
+  const interactionTracker = InteractionTracker.instance;
+  let incrementTaskCountSpy: jasmine.Spy;
+  let decrementTaskCountSpy: jasmine.Spy;
+  let onEndSpanSpy: jasmine.Spy;
+
+  beforeEach(() => {
+    interactionTracker.resetCurrentTracingZone();
+    incrementTaskCountSpy = spyOn(
+      interactionTracker,
+      'incrementTaskCount'
+    ).and.callThrough();
+    decrementTaskCountSpy = spyOn(
+      interactionTracker,
+      'decrementTaskCount'
+    ).and.callThrough();
+    onEndSpanSpy = spyOn(tracing.exporter, 'onEndSpan');
+  });
+
+  it('should handle interactions with no async work', done => {
+    fakeInteraction(noop);
+
+    onEndSpanSpy.and.callFake((rootSpan: Span) => {
+      // Counts related to the click event.
+      expect(incrementTaskCountSpy.calls.count()).toBe(1);
+      expect(decrementTaskCountSpy.calls.count()).toBe(1);
+      expect(rootSpan.name).toBe('test interaction');
+      expect(rootSpan.ended).toBeTruthy();
+      // As there is another setTimeOut that completes the interaction, the
+      // span duraction is not precise, then only test if the interaction duration
+      // finishes within a range.
+      expect(rootSpan.duration).toBeLessThan(5);
```

Comment:
Should you check a minimum time here too?

Reply:
In this case, this noop sync function does not have any logic.Then, the minimum time would be 0. I think that just checking the max time is fine.